app/dv/src/main/java/io/syndesis/dv/openshift/SyndesisConnectionSynchronizer.java

```diff
@@ -121,18 +121,34 @@ public void addConnection(DefaultSyndesisDataSource sds, boolean update) {
             return;
         }
 
+        boolean create = true;
         if (update) {
             try {
-                this.openshiftClient.deleteDataSource(sds);
+                //ensure that the properties have changed
+                if (sds.getTeiidName() != null) {
+                    TeiidDataSource tds = this.metadataService.findTeiidDatasource(sds.getTeiidName());
+                    if (tds != null) {
+                        DefaultSyndesisDataSource existing = tds.getSyndesisDataSource();
+                        if (existing.getProperties().equals(sds.getProperties())) {
```

Comment:
Is Object equality is what you are going for here?

Reply:
Yes.  Do you have a concern?